#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_cms_linkage.dpatch by Tim Retout <tim@retout.co.uk>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix from upstream 0.4.2 stable branch (rev 3692).
## DP: Fix linking Cms.dll against libfspot.

@DPATCH@
diff -urNad f-spot~/configure.in f-spot/configure.in
--- f-spot~/configure.in	2008-03-09 11:12:45.000000000 +0000
+++ f-spot/configure.in	2008-03-09 11:19:51.000000000 +0000
@@ -292,6 +292,7 @@
 src/Core/Defines.cs
 src/AssemblyInfo.cs
 src/f-spot.exe.config
+src/Cms.dll.config
 src/Makefile
 glitz-sharp/Makefile
 glitz-sharp/src/Makefile
diff -urNad f-spot~/src/Cms.dll.config f-spot/src/Cms.dll.config
--- f-spot~/src/Cms.dll.config	2008-03-09 11:12:45.000000000 +0000
+++ f-spot/src/Cms.dll.config	1970-01-01 01:00:00.000000000 +0100
@@ -1,4 +0,0 @@
-<configuration>
-  <dllmap dll="liblcms-1.0.0.dll" target="liblcms.so.1"/>
-</configuration>
-
diff -urNad f-spot~/src/Cms.dll.config.in f-spot/src/Cms.dll.config.in
--- f-spot~/src/Cms.dll.config.in	1970-01-01 01:00:00.000000000 +0100
+++ f-spot/src/Cms.dll.config.in	2008-03-09 11:19:51.000000000 +0000
@@ -0,0 +1,5 @@
+<configuration>
+  <dllmap dll="liblcms-1.0.0.dll" target="liblcms.so.1"/>
+  <dllmap dll="libfspot" target="@expanded_libdir@/f-spot/libfspot.so.0"/>
+</configuration>
+
diff -urNad f-spot~/src/Makefile.am f-spot/src/Makefile.am
--- f-spot~/src/Makefile.am	2008-03-09 11:12:45.000000000 +0000
+++ f-spot/src/Makefile.am	2008-03-09 11:19:51.000000000 +0000
@@ -384,7 +384,7 @@
 	$(F_SPOT_DISTRESOURCES)			\
 	f-spot.exe.config.in			\
 	f-spot.in				\
-	Cms.dll.config
+	Cms.dll.config.in
 
 CLEANFILES =					\
 	f-spot.exe.config			\
@@ -398,7 +398,8 @@
 	FSpot.Query.dll				\
 	FSpot.Query.dll.mdb			\
 	FSpot.Core.dll.mdb			\
-	FSpot.Core.dll
+	FSpot.Core.dll				\
+	Cms.dll.config
 
 
 DISTCLEANFILES = 				\
